spec.html

```diff
@@ -34832,14 +35211,31 @@ <h1>Notation</h1>
           <li>
             _Unicode_ is *true* if the RegExp object's [[OriginalFlags]] internal slot contains *"u"* and otherwise is *false*.
           </li>
+          <li>
+            _UnicodeSets_ is *true* if the RegExp object's [[OriginalFlags]] internal slot contains *"v"* and otherwise is *false*.
+          </li>
+          <li>
+            _EitherUnicode_ is *true* if _Unicode_ is *true* or _UnicodeSets_ is *true*, and otherwise is *false*.
+          </li>
           <li oldids="sec-runtime-semantics-wordcharacters-abstract-operation">
-            _WordCharacters_ is the mathematical set that is the union of all sixty-three characters in *"ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789_"* (letters, numbers, and U+005F (LOW LINE) in the Unicode Basic Latin block) and all characters _c_ for which _c_ is not in that set but Canonicalize(_c_) is. _WordCharacters_ cannot contain more than sixty-three characters unless _Unicode_ and _IgnoreCase_ are both *true*.
+            _WordCharacters_ is the mathematical set that is the union of all sixty-three characters in *"ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789_"* (letters, numbers, and U+005F (LOW LINE) in the Unicode Basic Latin block) and all characters _c_ for which _c_ is not in that set but Canonicalize(_c_) is. _WordCharacters_ cannot contain more than sixty-three characters unless _EitherUnicode_ and _IgnoreCase_ are both *true*.
           </li>
         </ul>
         <p>Furthermore, the descriptions below use the following internal data structures:</p>
         <ul>
           <li>
-            A <em>CharSet</em> is a mathematical set of characters. When the _Unicode_ flag is *true*, &ldquo;all characters&rdquo; means the CharSet containing all code point values; otherwise &ldquo;all characters&rdquo; means the CharSet containing all code unit values.
+            A <em>CharSetElement</em> is one of the two following entities:
+            <ul>
+              <li>
+                If _UnicodeSets_ is *false*, then a CharSetElement is a character in the sense of the Pattern Semantics above.
+              </li>
+              <li>
+                If _UnicodeSets_ is *true*, then a CharSetElement is either a character in the sense of the Pattern Semantics above, or it is a sequence of characters, that is, a string. This includes the empty String and strings with more than 1 character. A string of length 1 is the same as a single character.
```

Comment:
Depending on the flags, it's a set of code units, or code points, or code points plus sequences of code points. The concept of "character" under the new flag encompasses single code points but is also sufficiently general for the logical concept of "character" which frequently includes things that are encoded in Unicode with sequences of multiple code points. (Of course not vice versa.)

Also, renaming it would affect parts of the spec that need not be touched for the normative changes here.

There may or may not be a better name for this. If we find one, can we make that change later, on its own (separate from these normative changes)?

Reply:
This name is confusing enough that I would be quite uncomfortable landing it unless we've though looked hard for better names and failed to find one. I'll raise this at the editor call next week.

Reply:
Editors are agreed this will need a new name before landing the PR. That doesn't need to block anything until that point, though; it can still get stage 3.